packages/shared/src/components/auth/EmailCodeVerification.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { Button } from '../buttons/Button';
+import { TextField } from '../fields/TextField';
+import { AuthFormProps } from './common';
+import { AuthFlow } from '../../lib/kratos';
+import useAccountEmailFlow from '../../hooks/useAccountEmailFlow';
+import AuthForm from './AuthForm';
+import { KeyIcon, MailIcon, VIcon } from '../icons';
+import { AuthEventNames } from '../../lib/auth';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+
+interface EmailCodeVerificationProps extends AuthFormProps {
+  code?: string;
+  email: string;
+  flowId: string;
+  onSubmit?: () => void;
+  className?: string;
+}
+function EmailCodeVerification({
+  code: codeProp,
+  email: emailProp,
+  flowId,
+  onSubmit,
+  className,
+}: EmailCodeVerificationProps): ReactElement {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [hint, setHint] = useState('');
+  const [alert, setAlert] = useState({ firstAlert: true, alert: false });
```

Comment:
I find these alert and firstAlert names really confusing to understand what they are about 😅